### PR TITLE
fix formatting issues with Reshape documentation

### PR DESCRIPTION
**Pull request by nswamy**
fix formatting issues with Reshape, preview can be seen in the screenshot

@mli, @zackchase, @Roshrini @madjam

[Reshape.pdf](https://github.com/dmlc/mxnet/files/894397/Reshape.pdf)

### CODE REVIEW

**Comment by piiswrong** [comment]
I thought this was merged already

**Comment by nswamy** [comment]
@piiswrong it was merged but had to revert it, formatting of the text was a little messed up.

**Comment by mli** [comment]
why not just make a PR to fix the error instead of reverting and then fixing.

**Comment by nswamy** [comment]
I made a pull request yesterday to revert the broken formatting. Fixed it today(updated the PR)

**Comment by nswamy** [comment]
Changes to  abs,sign,round,rint,ceil,floor,rint,fix,square,sqrt,rsqrt,exp,log,log10,log10,log2,sin,log1p,cos math function documentation
Please take a look
@mli, @zackchase, @Roshrini @madjam

File: python/mxnet/ndarray_doc.py [hunk 1]
```diff
@@ -3,12 +3,39 @@
...
     """The basic class"""
     pass
 
+class ReshapeDoc(NDArrayDoc):
+    """
+    Examples
+    --------
+    Reshapes the input array into a new shape.
+
+    >>> x = mx.nd.array([1,2,3,4])
```
Comment (madjam) [inline python/mxnet/ndarray_doc.py]: Always add a space after comma

File: src/operator/tensor/matrix_op.cc [hunk 2]
```diff
@@ -23,57 +23,66 @@ DMLC_REGISTER_PARAMETER(ReverseParam);
 
 NNVM_REGISTER_OP(Reshape)
 .add_alias("reshape")
-.describe(R"code(Reshape array into a new shape.
+.describe(R"code(Reshapes the input array into a new shape.
```
Comment (madjam) [inline src/operator/tensor/matrix_op.cc]: Should array be changed to ndarray here and in a few other places?

File: src/operator/tensor/elemwise_unary_op.cc [hunk 3]
```diff
@@ -314,7 +344,7 @@ MXNET_OPERATOR_REGISTER_BINARY(_backward_expm1)
 MXNET_OPERATOR_REGISTER_UNARY(cos)
 .describe(R"code(Cosine, element-wise.
```
Comment (madjam) [inline src/operator/tensor/elemwise_unary_op.cc]: Should the text here be similar to what you have for `sin`?

File: src/operator/tensor/elemwise_unary_op.cc [hunk 4]
```diff
@@ -90,7 +93,7 @@ For example::
   })
 .set_attr<FCompute>("FCompute<cpu>", CastCompute<cpu>)
 .set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseNone{"_backward_cast"})
-.add_argument("data", "NDArray-or-Symbol", "Source input")
+.add_argument("data", "ndarray-or-symbol", "The input.")
```
Comment (zackchase) [inline src/operator/tensor/elemwise_unary_op.cc]: Change to "NDArray-or-Symbol" <- should be capitalized to be consistent with convention in Python docs.

File: src/operator/tensor/elemwise_unary_op.cc [hunk 5]
```diff
@@ -90,7 +93,7 @@ For example::
...
 .add_arguments(CastParam::__FIELDS__());
```
Comment (mli) [inline src/operator/tensor/elemwise_unary_op.cc]: `NDArray-or-Symbol` is required, should not change it

File: src/operator/tensor/matrix_op.cc [hunk 6]
```diff
@@ -23,57 +23,66 @@ DMLC_REGISTER_PARAMETER(ReverseParam);
...
+  - input shape = (2,3,4), shape = (-3,-2), output shape = (6,4)
+
+- ``-4`` split one dimension of the input into two dimensions passed subsequent to -4 in shape (can contain -1).
+
+  Example::
+
+  - input shape = (2,3,4), shape = (-4,1,2,-2), output shape =(1,2,3,4)
+  - input shape = (2,3,4), shape = (2,-4,-1,3,-2), output shape = (2,1,3,4)
+
+If the argument ``reverse`` is set to 1, then the special values are inferred from right to left.
```
Comment (zackchase) [inline src/operator/tensor/matrix_op.cc]: reverse is an argument name. Should be in single backticks.

File: src/operator/tensor/elemwise_unary_op.cc [hunk 7]
```diff
@@ -198,10 +214,15 @@ MXNET_OPERATOR_REGISTER_BINARY(_backward_square)
 
 // sqrt
 MXNET_OPERATOR_REGISTER_UNARY(sqrt)
-.describe(R"code(Calculate the square-root of an array, element-wise.
+.describe(R"code(Returns element-wise square-root value of the input.
 
-For example::
+.. math::
    sqrt(x) = \sqrt{x}
```
Comment (mli) [inline src/operator/tensor/elemwise_unary_op.cc]: may be change sqrt to \textrm{sqrt}

File: src/operator/tensor/elemwise_unary_op.cc (hunk 3 above)
Comment (nswamy) [inline src/operator/tensor/elemwise_unary_op.cc]: I haven't worked on all the math functions, i'll have similar text

File: src/operator/tensor/elemwise_unary_op.cc (hunk 4 above)
Comment (nswamy) [inline src/operator/tensor/elemwise_unary_op.cc]: done

File: src/operator/tensor/matrix_op.cc (hunk 6 above)
Comment (nswamy) [inline src/operator/tensor/matrix_op.cc]: done.

File: src/operator/tensor/elemwise_unary_op.cc (hunk 5 above)
Comment (nswamy) [inline src/operator/tensor/elemwise_unary_op.cc]: restored to NDArray-or-Symbol

File: src/operator/tensor/elemwise_unary_op.cc (hunk 7 above)
Comment (mli) [inline src/operator/tensor/elemwise_unary_op.cc]: \textrm{sqrt}(x) = \sqrt{x}

File: python/mxnet/ndarray_doc.py (hunk 1 above)
Comment (nswamy) [inline python/mxnet/ndarray_doc.py]: done

File: src/operator/tensor/matrix_op-inl.h [hunk 8]
```diff
@@ -32,18 +32,18 @@ struct ReshapeParam : public dmlc::Parameter<ReshapeParam> {
...
     DMLC_DECLARE_FIELD(target_shape)
     .set_default(TShape(tmp, tmp + 2))
     .describe("(Deprecated! Use ``shape`` instead.) "
               "Target new shape. One and only one dim can be 0, "
               "in which case it will be inferred from the rest of dims");
+    DMLC_DECLARE_FIELD(keep_highest).set_default(false)
+    .describe("(Deprecated! Use ``shape`` instead.) Whether keep the highest dim unchanged."
+              "If set to true, then the first dim in target_shape is ignored,"
+              "and always fixed as input");
   }
```
Comment (eric-haibin-lin) [inline src/operator/tensor/matrix_op-inl.h]: Please don't change the order of of fields declared by `DMLC_DECLARE_FIELD `. `DMLC_DECLARE_FIELD(keep_highest).set_default(false)` should be followed by `DMLC_DECLARE_FIELD(shape)`.
The order determines the auto-generated cpp interface for the operator. This change breaks the cpp-package build.

@nswamy I'm going to make a PR to revert this change, and add a CI job for cpp-package.
Also ping @zackchase @Roshrini @jiajiechen  for awareness.

**Pull request merged**